### PR TITLE
Refactor vm_setup to use struct for nics

### PR DESCRIPTION
**Pull request by furkansahin**

### CODE REVIEW

File: lib/hosting/hetzner_apis.rb [hunk 1]
```diff
@@ -2,8 +2,6 @@
 
 require "excon"
 class Hosting::HetznerApis
-  FailoverSubnet = Struct.new(:ips, :subnets, :failovers)
```
Comment (fdr) [inline lib/hosting/hetzner_apis.rb:5]: Getting rid of this is unrelated right?

**Review by fdr — COMMENTED** [review]
Looks pretty good, I like the size decrease.

File: lib/hosting/hetzner_apis.rb (hunk 1 above)
Comment (furkansahin) [reply lib/hosting/hetzner_apis.rb:5]: yup, just realized it's never used, considering the size of the PR, wanted to squish that in

Comment (fdr) [reply lib/hosting/hetzner_apis.rb:5]: I'd just shave it off into its own commit, though you can stuff it in the PR. I am generally more relaxed about PR metadata, since I really only load it when I have to do doubly-invasive root-cause analysis.

**Review by fdr — APPROVED** [review]

**Comment by fdr** [comment]
I suppose it would be nice if you briefly explained the refactor for the first one, so someone running `git log` and not `git log -p` (plus no patience to wade into the code) can understand the status quo ante (it used to be an array of four elements addressed by position)

**Comment by fdr** [comment]
But, no big deal I guess
